hammer_time/hammer_time.py

```diff
@@ -140,6 +141,34 @@ def perform(cls, client, machine_id):
         ctrl_client.juju('ssh', (machine_id,) + cls.kill_script)
 
 
+class InterruptNetworkAction(MachineAction):
+
+    def get_command():
+        deny_all = '; '.join([
+            'iptables --flush',
+            'iptables -P FORWARD DROP',
+            'iptables -P OUTPUT DROP',
+            'iptables -P INPUT DROP',
+            ])
+        restore = 'iptables-restore < $HOME/iptables'
+        commands = [
+            'set -eux',
+            'sudo iptables-save > $HOME/iptables',
+            'echo "{}"| sudo at now + 6 minutes'.format(restore),
+            'echo "{}"| sudo at now + 1 minutes'.format(deny_all),
+            ]
+        return '; '.join(commands)
```

Comment:
I don't think the risk of saving bad rules is very high.  This is done immediately after connecting via ssh, so whatever rules we save necessarily permit ssh access.  If we have ssh, we certainly have enough to repair.

I don't think there is a scenario where InterruptNetworkAction is called again after it has already failed.  The first failure should terminate Hammer Time.  Yes, we could re-run Hammer Time to invoke InterruptNetworkAction again, but 
- Hammer Time is not supposed to support that.  Don't do that.
- The wait_for_started should notice that the machine is down, and so refuse to run anything else.

Because Juju uses port 22, I don't want it to remain active.  Some providers will let us do an autopsy without ssh (e.g. lxc exec, or rebooting.)    I could maybe see using a different port for ssh, e.g. 8022, so that we can do an autopsy, but I'd rather wait until we know that we need it, because this stuff is hard.  Just getting `iptables` & `at` to work has taken the better part of a day.  Port forwarding would be several more hours, I believe.

Another interesting option would be to allow access (to everything!) for a specific IP address, e.g. ci-master.

I think your rule would not permit ssh access, because it doesn't allow output.